packages/fuels/tests/harness.rs

```diff
@@ -4180,3 +4180,119 @@ async fn test_vector() -> Result<(), Error> {
+        TxParameters::default(),
+        StorageConfiguration::default(),
+    )
+    .await?;
+
+    let contract_instance = MyContract::new(contract_id.to_string(), wallets[0].clone());
+    let contract_methods = contract_instance.methods();
+    let mint_asset_id = AssetId::from(*contract_id.hash());
+    let addresses: [Address; 3] = wallets
+        .iter()
+        .map(|wallet| wallet.address().into())
+        .collect::<Vec<Address>>()
+        .try_into()
+        .unwrap();
+    let amount = 1000;
+
+    {
+        // Should fail due to lack of output variables
+        let response = contract_methods
+            .mint_to_addresses(amount, addresses)
+            .call()
+            .await;
+
+        assert!(matches!(response, Err(Error::RevertTransactionError(..))));
+    }
+
+    {
+        // Should fail due to insufficient attempts (needs at least 3)
+        let response = contract_methods
+            .mint_to_addresses(amount, addresses)
+            .estimate_tx_dependencies(Some(2))
+            .await;
+
+        assert!(matches!(response, Err(Error::RevertTransactionError(..))));
+    }
+
+    {
+        // Should add 3 output variables automatically
+        let _ = contract_methods
+            .mint_to_addresses(amount, addresses)
+            .estimate_tx_dependencies(Some(3))
+            .await?
+            .call()
+            .await?;
+
+        for wallet in wallets.iter() {
+            let balance = wallet.get_asset_balance(&mint_asset_id).await?;
+            assert_eq!(balance, amount);
+        }
+    }
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_output_variable_estimation_multicall() -> Result<(), Error> {
+    abigen!(
+        MyContract,
+        "packages/fuels/tests/test_projects/token_ops/out/debug/token_ops-abi.json"
+    );
+
+    let wallet_config = WalletsConfig::new(Some(3), None, None);
+    let wallets = launch_custom_provider_and_get_wallets(wallet_config, None).await;
+
+    let contract_id = Contract::deploy(
+        "tests/test_projects/token_ops/out/debug/token_ops.bin",
+        &wallets[0],
+        TxParameters::default(),
+        StorageConfiguration::default(),
+    )
+    .await?;
+
+    let contract_instance = MyContract::new(contract_id.to_string(), wallets[0].clone());
+    let contract_methods = contract_instance.methods();
+    let mint_asset_id = AssetId::from(*contract_id.hash());
+    let addresses: [Address; 3] = wallets
+        .iter()
+        .map(|wallet| wallet.address().into())
+        .collect::<Vec<Address>>()
+        .try_into()
+        .unwrap();
+    let amount = 1000;
+
+    let mut multi_call_handler = MultiContractCallHandler::new(wallets[0].clone());
+    (0..3).for_each(|_| {
+        let call_handler = contract_methods.mint_to_addresses(amount, addresses);
+        multi_call_handler.add_call(call_handler);
+    });
+
+    let _ = multi_call_handler
+        .estimate_tx_dependencies(None)
+        .await?
+        .call::<((), (), ())>()
```

Comment:
Is this the syntax the end-user would have to use in multi-call contexts? Can we make it better? I feel like it could be abstracted away, but maybe it involves something too complicated.

Reply:
This is would be a prettier alternative: 

```rust
let (counter, array): (u64, [u64; 2]) = multi_call_handler.call().await?.value;
```
The user must, at some point, define what types they are expecting as return. We spent quite some time weighing different options when this was developed and concluded this was the friendliest.